### PR TITLE
Feat/google

### PR DESCRIPTION
**Pull request by MargaritaSukhova**
## Describe your changes

- Add Google sign up and sign in
- Refactor reducers and user api service and combined them to one slice and user api service
- Changed validation in sign up form to Yup resolver and add validation schema
- Finished change password flow and add Access token to user api service for protected backend roots
- Updated styles for disabled state of styled button

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](#https://ticket-url)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
Снимок экрана 2024-06-12 в 22 17 36
Снимок экрана 2024-06-12 в 22 17 50
Снимок экрана 2024-06-12 в 22 18 40
Снимок экрана 2024-06-12 в 22 18 48
Снимок экрана 2024-06-12 в 22 20 18
Снимок экрана 2024-06-12 в 22 31 30

### CODE REVIEW

File: src/components/GoogleLoginButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            }
+          } else {
+            console.error('Response data is undefined');
+          }
+        } catch (err) {
+          alert(err);
+        }
+      }}
+      onError={() => {
+        console.log('Login Failed');
```
Comment (lyudad) [inline src/components/GoogleLoginButton/index.tsx]: ?

File: src/pages/NewPasswordPage/NewPasswordForm/index.tsx [hunk 2]
```diff
@@ -50,24 +44,18 @@ function NewPasswordForm() {
...
-        password: data.password,
-        repeatPassword: data.repeatPassword,
-      }).unwrap();
-
-      dispatch(resetPasswordSuccess(response));
+      await newPassword({ password }).unwrap();
+      showToast('success', t('newPassword.passwordChanged'));
+      navigate(urls.HOME);
     } catch (err) {
+      console.log(err);
```
Comment (lyudad) [inline src/pages/NewPasswordPage/NewPasswordForm/index.tsx]: ?

File: src/redux/user/types.ts [hunk 3]
```diff
@@ -1,32 +1,21 @@
...
   isEmailVerified: boolean;
   isLoggedIn: boolean;
-  accessToken: string;
-  refreshToken: string;
-  error: string;
-}
-
-export interface IUserState {
-  user: IUser;
+  accessToken: string | undefined;
```
Comment (lyudad) [inline src/redux/user/types.ts]: accessToken?: string;

Comment (MargaritaSukhova) [reply src/redux/user/types.ts]: thank you, fixed it

File: src/pages/NewPasswordPage/NewPasswordForm/index.tsx (hunk 2 above)
Comment (MargaritaSukhova) [reply src/pages/NewPasswordPage/NewPasswordForm/index.tsx]: deleted console

File: src/components/GoogleLoginButton/index.tsx (hunk 1 above)
Comment (MargaritaSukhova) [reply src/components/GoogleLoginButton/index.tsx]: fixed it

**Review by lyudad — APPROVED** [review]

**Pull request merged**